river/sync.go

```diff
@@ -385,7 +395,27 @@ func (r *River) makeInsertReqData(req *elastic.BulkRequest, rule *Rule, values [
 			mysql, elastic, fieldType := r.getFieldParts(k, v)
 			if mysql == c.Name {
 				mapped = true
-				req.Data[elastic] = r.getFieldValue(&c, fieldType, values[i])
+				if fieldType == fieldTypeGeopointLat {
```

Comment:
why not handing this in getFieldValue?

Reply:
because getFieldValue has no access to already read field values and would overwrite the first part of the geo point.
for long and lat it would do:
1. getFieldValue for longitude. returns map with { long: n }
2. getFieldValue for latitude. return map with { lat: n }
3. writing data would require to merge the both results, because we can't access the already processed data (or next)